### ISSUE TITLE
AWS I/O operation timeout setting on Debian Jessie Image with NVMe based VMs (eg. c5, m5s) and AWS EBS volumes

### ISSUE THREAD

**Issue by ottoyiu** [kubernetes/kops#5694]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Version 1.9.0 (git-cccd71e67)

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

We're running `kope.io/k8s-1.9-debian-jessie-amd64-hvm-ebs-2018-03-11` as our image.

**3. What cloud provider are you using?**
AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
We're running into I/O operation timeouts with our c5's and a NVMe EBS root volume, which lead to the root filesystem to be mounted as read-only.

```
Aug 16 04:57:44 ip-10-67-98-200 kernel: nvme nvme0: I/O 29 QID 1 timeout, aborting
Aug 16 04:57:46 ip-10-67-98-200 kernel: nvme nvme0: I/O 0 QID 1 timeout, aborting
Aug 16 04:57:46 ip-10-67-98-200 kernel: nvme nvme0: I/O 1 QID 1 timeout, aborting
Aug 16 04:57:46 ip-10-67-98-200 kernel: nvme nvme0: I/O 2 QID 1 timeout, aborting
Aug 16 04:57:46 ip-10-67-98-200 kernel: nvme nvme0: I/O 3 QID 1 timeout, aborting
```

AWS suggests, when dealing with NVMe based EBS volumes to bump up the `nvme_core.io_timeout` and `nvme_core.max_retries` in the kernel boot parameter.
```
I/O Operation Timeout
NVMe EBS volumes use the default NVMe driver provided by the operating system. Most operating systems specify a timeout for I/O operations submitted to NVMe devices. The default timeout is 30 seconds and can be changed using the nvme_core.io_timeout boot parameter (or the nvme.io_timeout boot parameter for Linux kernels prior to version 4.6). For an experience similar to EBS volumes attached to Xen instances, we recommend setting this to the highest value possible. For Amazon Linux AMI 2017.09.01 (or greater), and for Linux kernels with version 4.15 or greater, the maximum is 4294967295. Prior to Linux 4.15, the maximum is 255 seconds. If you are using a current version of the Amazon Linux AMI, we have already increased the timeout.

With Linux kernel 4.14 and later, you can also configure the number of times that I/O operations can be retried. The default is five retries. You can configure a different value using the nvme_core.max_retries kernel boot parameter, or at runtime using the following command:
```
Source: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html

It would be nice to be able to bake this into the image itself instead for us to run this ourselves:
```
cp /etc/default/grub /tmp/grub
cat >>/tmp/grub <<'EOF'
GRUB_CMDLINE_LINUX="${GRUB_CMDLINE_LINUX} nvme_core.io_timeout=255"
EOF
sudo mv /tmp/grub /etc/default/grub
sudo update-grub
```

Is there a repo where we can contribute for this base image?
I guess it's here: https://github.com/kubernetes/kube-deploy/tree/master/imagebuilder

Thanks!

**Comment by ottoyiu** [comment]
I think we need to change this line:
https://github.com/kubernetes/kube-deploy/blob/36dbd4f085acb0a052cafdbb012bb26e329d50e0/imagebuilder/templates/1.10-jessie.yml#L182

**Comment by ottoyiu** [comment]
/remove-lifecycle stale

@ercliou have you had the chance to look at that PR comment by justinsb?

**Comment by djagannath-asapp** [comment]
We ran into this issue on several m5 instances. We're running an Ubuntu AMI from January 2018.  Looks like it was fixed in AMI's newer than 20180415. https://bugs.launchpad.net/ubuntu/bionic/+source/linux/+bug/1758466
